containers-tests/tests/intmap-properties.hs

```diff
@@ -1539,10 +1552,23 @@ instance Applicative Identity where
   pure a = Identity a
   Identity f <*> Identity a = Identity (f a)
 
-prop_traverseWithKey :: IntMap () -> Bool
-prop_traverseWithKey mp = mp == newMap
+prop_traverseWithKey_identity :: IntMap () -> Bool
+prop_traverseWithKey_identity mp = mp == newMap
   where Identity newMap = traverseWithKey (\_ -> Identity) mp
```

Comment:
You still don't have a test showing that `traverseWithKey` degrades to `mapWithKey`, which would be strictly stronger than this test. Also, this test only looks at keys, not values, so it's too weak even as an identity test.

Reply:
To have an `identity test`, the result would have to be the same as the input, right?
`traverseWithKey :: Applicative t => (Key -> a -> t b) -> IntMap a -> t (IntMap b)`
given that `a ~ b` if this is an identity test, how would you use `Key` to make an `a` in the first argument? (`Key -> a -> t a`)

I'll add the `traverseWithKey` to `mapWithKey` degredation test tomorrow.

Reply:
My point is that the degradation test plus the identity test for what's degraded to effectively give you an identity test